test/caching-precompiler.js

```diff
@@ -18,9 +20,14 @@ function endsWithMap(filename) {
 	return /\.js$/.test(filename);
 }
 
+test('before', function (t) {
+	sinon.spy(babel, 'transform');
```

Comment:
AVA's tests are run using [`tap`](http://www.node-tap.org/) which runs each test file in its own process. You don't have to worry about modifying the `babel-core` module, or about cleaning it up afterwards. Just add the spy in the module itself, outside of a test.